static/js/copy_and_paste.js

```diff
@@ -96,44 +124,80 @@ function copy_handler() {
             return row.next();
         });
         if (start_data === undefined) {
-            return;
+            // Skip any selection sections that don't intersect a message.
+            continue;
+        }
+        if (start_id === undefined) {
+            // start_id is the Zulip message ID of the first message
+            // touched by the selection.
+            start_id = start_data[0];
         }
-        start_id = start_data[0];
 
         endc = $(range.endContainer);
-        // If the selection ends in the bottom whitespace, we should act as
-        // though the selection ends on the final message
-        // Chrome seems to like selecting the compose_close button
-        // when you go off the end of the last message
+        // If the selection ends in the bottom whitespace, we should
+        // act as though the selection ends on the final message.
+        // This handles the issue that Chrome seems to like selecting
+        // the compose_close button when you go off the end of the
+        // last message
         if (endc.attr('id') === "bottom_whitespace" || endc.attr('id') === "compose_close") {
             initial_end_tr = $(".message_row:last");
+            // The selection goes off the end of the message feed, so
+            // this is a multi-message selection.
             skip_same_td_check = true;
         } else {
             initial_end_tr = $(endc.parents('.selectable_row')[0]);
         }
         end_data = find_boundary_tr(initial_end_tr, function (row) {
             return row.prev();
         });
+
         if (end_data === undefined) {
-            return;
+            // Skip any selection sections that don't intersect a message.
+            continue;
+        }
+        if (end_data[0] !== undefined) {
+            end_id = end_data[0];
         }
-        end_id = end_data[0];
 
         if (start_data[1] || end_data[1]) {
+            // If the find_boundary_tr call for either the first or
+            // the last message covered by the selection
             skip_same_td_check = true;
         }
 
-        // we should let the browser handle the copy-paste entirely on its own
-        // (In this case, there is no need for our special copy code)
-        if (!skip_same_td_check &&
-            startc.parents('.selectable_row>div')[0] === endc.parents('.selectable_row>div')[0]) {
-            return;
+        if (skip_same_td_check) {
+            // Even if once it is checked that skip_same_td_check is true
+            // during the loop, we can skip the same td check for the
+            // whole selection
+            skip_same_td_check_selection = true;
```

Comment:
I still don't understand -- can't we just delete this variable and have the below code use `if (!skip_same_td_check && ...`?  You should console.log debug this -- I don't see any situation where those two would ever have different values at the end of this loop.

Reply:
Sorry @timabbott my bad. I didn't notice that `skip_same_td_check` was out of scope of loop.